overlord/snapstate/snapmgr.go

```diff
@@ -560,6 +660,12 @@ func (m *SnapManager) Ensure() error {
 
 	m.runner.Ensure()
 
+	// needs to run after the runner
+	if err := m.ensureSnapConfineApparmor(); err != nil {
+		// FIMXE: this will spam the world as ensure runs often
```

Comment:
Why are we doing that non-stop? I mean reading/writing/updating the profile, rather than just logging. That sounds unnecessary and somewhat unfortunate to be running so frequently. Shouldn't we be doing that only when updating core, which is the moment when we update snap-confine itself and the apparmor profile that comes with it?